spring-context/src/main/java/org/springframework/scheduling/config/ScheduledTaskRegistrar.java

```diff
@@ -92,9 +92,8 @@ public void setTaskScheduler(TaskScheduler taskScheduler) {
 	 * {@link java.util.concurrent.ScheduledExecutorService} to be wrapped as a
 	 * {@code TaskScheduler}.
 	 */
-	public void setScheduler(Object scheduler) {
-		Assert.notNull(scheduler, "Scheduler object must not be null");
-		if (scheduler instanceof TaskScheduler) {
+	public void setScheduler(@Nullable Object scheduler) {
+		if ((scheduler instanceof TaskScheduler) || (scheduler == null)) {
```

Comment:
We could code this as a dedicated `if` clause: `if (scheduler == null) { this.scheduler = null } else if (scheduler instanceof TaskScheduler) ...`